### PR TITLE
Make branch protection check work on non enterprise

### PR DESCRIPTION
**Pull request by schosterbarak**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Enable github configuration rules on non "github enterprise organization". 
AKA enable verifying branch protection rules on individual accounts. 

Fixes # BCD-239

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/vcs/base_vcs_dal.py [hunk 1]
```diff
@@ -58,14 +58,14 @@ def setup_http_manager(self, ca_certificate: str | None = None) -> None:
             except KeyError:
                 self.http = urllib3.PoolManager()
 
-    def _request(self, endpoint: str) -> dict[str, Any] | None:
+    def _request(self, endpoint: str, allowed_status_codes: list[int]) -> dict[str, Any] | None:
```
Comment (achiar99) [inline checkov/common/vcs/base_vcs_dal.py:61]: i think you need to init it with default value (but not inside the func signature). some other classes (like gitlab) also call this function

File: tests/github/resources/github_conf/repo_no_rules/branch_protection_rules.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "message": "Branch not protected",
```
Comment (achiar99) [inline tests/github/resources/github_conf/repo_no_rules/branch_protection_rules.json:2]: this is the response whit status code 404?

File: checkov/github/base_github_branch_security.py [hunk 3]
```diff
@@ -26,6 +27,10 @@ def scan_entity_conf(self, conf):
                 return CheckResult.PASSED
             else:
                 return CheckResult.FAILED
+        if no_branch_security_schema.validate(conf):
+            message = conf.get('message', '')
+            if message == 'Branch not protected':
```
Comment (achiar99) [inline checkov/github/base_github_branch_security.py:32]: i would insert this text into const

**Review by achiar99 — COMMENTED** [review]
Very nice!
small comments

File: tests/github/resources/github_conf/repo_no_rules/branch_protection_rules.json (hunk 2 above)
Comment (schosterbarak) [reply tests/github/resources/github_conf/repo_no_rules/branch_protection_rules.json:2]: yes.

**Review by achiar99 — APPROVED** [review]

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**